python/src/kagent/models/vertexai/_anthropic_vertex_client.py

```diff
@@ -0,0 +1,121 @@
+
+    if disallowed_create_args.intersection(create_args_keys):
+        raise ValueError(f"Disallowed create args are present: {disallowed_create_args.intersection(create_args_keys)}")
+
+    return create_args
+
+
+def _anthropic_client_from_config(config: Mapping[str, Any]) -> AsyncAnthropicVertex:
+    # Filter config to only include valid parameters
+    client_config = {k: v for k, v in config.items() if k in anthropic_init_kwargs}
+    return AsyncAnthropicVertex(**client_config)
+
+
+class AnthropicVertexAIChatCompletionClient(
+    BaseAnthropicChatCompletionClient, Component[AnthropicVertexAIClientConfiguration]
+):
+    component_type = "model"
+    component_config_schema = AnthropicVertexAIClientConfiguration
+    component_provider_override = "kagent.models.vertexai.AnthropicVertexAIChatCompletionClient"
+
+    def __init__(self, **kwargs: Unpack[AnthropicVertexAIClientConfiguration]):
+        if "model" not in kwargs:
+            raise ValueError("model is required for AnthropicVertexAIChatCompletionClient")
+
+        self._raw_config: Dict[str, Any] = dict(kwargs).copy()
+        copied_args = dict(kwargs).copy()
+
+        model_info: Optional[ModelInfo] = None
+        if "model_info" in kwargs:
+            model_info = kwargs["model_info"]
+            del copied_args["model_info"]
+
+        if "credentials" in kwargs:
+            credentials = kwargs["credentials"]
+            del copied_args["credentials"]
+        else:
+            raise ValueError("credentials is required for AnthropicVertexAIChatCompletionClient")
+
+        if "project" in kwargs:
+            project = kwargs["project"]
+            del copied_args["project"]
+        else:
+            raise ValueError("project is required for AnthropicVertexAIChatCompletionClient")
+
+        if "location" in kwargs:
+            location = kwargs["location"]
+            del copied_args["location"]
+        else:
+            raise ValueError("location is required for AnthropicVertexAIChatCompletionClient")
+
+        # need to explicitly provide the scopes for the credentials, otherwise it will not work
+        google_creds = load_credentials_from_dict(
+            credentials, scopes=["https://www.googleapis.com/auth/cloud-platform"]
+        )
+
+        client = AsyncAnthropicVertex(
+            region=location,
+            project_id=project,
+            credentials=google_creds[0],
+        )
+        create_args = _create_args_from_config(copied_args)
+
+        super().__init__(
+            client=client,
+            create_args=create_args,
+            model_info=model_info,
+        )
+
+    def __getstate__(self) -> Dict[str, Any]:
+        state = self.__dict__.copy()
+        state["_client"] = None
+        return state
+
+    def __setstate__(self, state: Dict[str, Any]) -> None:
+        self.__dict__.update(state)
+        self._client = _anthropic_client_from_config(state["_raw_config"])
```

Comment:
What are these for?

Reply:
it's used for serailization -- I follow the same pattern as upstream implementation.